src/Uno.UI/UI/Xaml/Controls/FocusVisual/SystemFocusVisual.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using Uno.Disposables;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Uno.UI.Xaml.Controls
+{
+	public partial class SystemFocusVisual : Control
+	{
+		private SerialDisposable _focusedElementSubscriptions = new SerialDisposable();
+
+		public SystemFocusVisual()
+		{
+			DefaultStyleKey = typeof(SystemFocusVisual);
+		}
+
+		protected override void OnApplyTemplate() => base.OnApplyTemplate();
+
+		public FrameworkElement FocusedElement
+		{
+			get => (FrameworkElement)GetValue(FocusedElementProperty);
+			set => SetValue(FocusedElementProperty, value);
+		}
+
+		public static readonly DependencyProperty FocusedElementProperty =
+			DependencyProperty.Register(
+				nameof(FocusedElement),
+				typeof(FrameworkElement),
+				typeof(SystemFocusVisual),
+				new PropertyMetadata(default, OnFocusedElementChanged));
+
+		private static void OnFocusedElementChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+		{
+			var focusVisual = (SystemFocusVisual)dependencyObject;
+
+			focusVisual._focusedElementSubscriptions.Disposable = null;
+
+			if (args.NewValue is FrameworkElement element)
+			{
+				element.SizeChanged += focusVisual.FocusedElementSizeChanged;
+				element.LayoutUpdated += focusVisual.FocusedElemenLayoutUpdated;
+
+				focusVisual.SetLayoutProperties();
+
+				focusVisual._focusedElementSubscriptions.Disposable = Disposable.Create(() =>
```

Comment:
Make sure to set the disposable to null first, otherwise subscriptions may not get removed properly.

Reply:
I'm setting it null a bit above on line 36 👍